R/v3-sample-utils.R

```diff
@@ -6,58 +6,75 @@
 # compound task id sets across modeling tasks). This is achieved by using a full
 # join between tbl sample data and the model output data sample grid for each
 # modeling task. This means that only valid task id combinations are considered.
-spl_hash_tbl <- function(tbl, round_id, config_tasks) {
+spl_hash_tbl <- function(tbl, round_id, config_tasks, compound_taskid_set = NULL) {
   tbl <- tbl[tbl$output_type == "sample", names(tbl) != "value"]
 
   mt_spl_grid <- expand_model_out_grid(
     config_tasks = config_tasks,
     round_id = round_id,
     all_character = TRUE,
     include_sample_ids = TRUE,
-    bind_model_tasks = FALSE
-  )
+    bind_model_tasks = FALSE,
+    compound_taskid_set = compound_taskid_set
+  ) %>%
+    stats::setNames(seq_along(.))
 
-  spl_mts <- has_spls_mt_grid(mt_spl_grid)
+  mt_tbls <- purrr::map(
+    mt_spl_grid,
+    function(.x) {
+      if (is.null(.x) || !has_spls_tbl(.x)) {
+        NULL
+      } else {
+        dplyr::filter(
+          .x, .data$output_type == "sample"
+        ) %>%
+          dplyr::rename(compound_idx = "output_type_id") %>%
+          dplyr::inner_join(tbl, .,
+            by = setdiff(
+              names(tbl),
+              c("output_type_id", "compound_idx")
+            )
+          )
+      }
+    }
+  )
 
```

Comment:
The above is a refactoring of the below that allows us to not have to subset different modeling tasks but instead do nothing when there are no samples in a given modeling task. Overall this makes downstream step cleaner.